front/pages/api/v1/w/[wId]/apps/[aId]/runs/index.ts

```diff
@@ -174,6 +174,10 @@ async function handler(
   if (keyRes.isErr()) {
     return apiError(req, res, keyRes.error);
   }
+
+  // TODO(2024-08-02 flav) Refactor auth.fromKey logic.
+  // Confusingly, the auth workspace here is the the one from the URL, not the one from the key.
+  // Where as auth.groups are the groups associated with the the key.
```

Comment:
💯 to change the behaviour and keep in auth the workspace of the user key - keeping a different workspace here is very confusing. The workspace from the url is only there to know in which workspace is the app , but it has nothing to with authentication.

Reply:
We will always have two workspaces here as we can run an app from workspace A with a key from workspace B. This is tricky indeed and need thorough consideration in the context of groups 👍

Reply:
We will always have two workspaces here as we can run an app from workspace A with a key from workspace B. This is tricky indeed and needs thorough consideration in the context of groups 👍

Reply:
We do need an auth on the app workspace as this is the one we use to retrieve the app. Let's call this the appAuth.

That being said it would be nice to have an auth on the key independent of the app workspace. This one could be called keyAuth.

That way we would not need to fetch the groups and could pass the keyAuth directly to create run?